Source/WebKit/Platform/IPC/Connection.cpp

```diff
@@ -129,6 +129,34 @@ class Connection::SyncMessageState {
     Deque<ConnectionAndIncomingMessage> m_messagesToDispatchWhileWaitingForSyncReply WTF_GUARDED_BY_LOCK(m_lock);
 };
 
+// Connections created for background threads don't (yet) support receiving messages while waiting for a sync reply.
```

Comment:
I don't think this is a the best approach, it makes it overly complicated to understand if it's correct or not and overly complicated to understand how the sync messages and waited on messages are dispatched.

The wait here, I think, completes before the message has been dispatched. This is not what the system is trying to achieve.

I think better is to:
1. Make a ReentrantFunctionDispatcher that supports dispatchSync()
2. Make a mapping of ReentrantFunctionDispatcher -> SyncMessageState

Other alternative is to ensure Non-mainthread connections do not have non-complex received messages. E.g. RRBProxy could wait for a sync point for flush to complete, but then the "did create backend" message would need another solution.  This is not very general solution.

This, too, should be worked on separately and tests should be added.